File: src/main/java/com/github/cdefgah/poetica/model/EmailsCountDigest.java
/*
 * SPDX-License-Identifier: GPL-3.0-only
 * Copyright (c) 2020 by Rafael Osipov <rafael.osipov@outlook.com>
 */

package com.github.cdefgah.poetica.model;

/**
 * Используется для получения сводки по загруженным в систему письмам для той или иной команды.
 */
public final class EmailsCountDigest {

    /**
     * Количество писем, присланных на предварительный тур (раунд).
     */
    private final long emailsQtyForTheFirstRound;

    /**
     * Количество писем, присланных на основной тур (раунд).
     */
    private final long emailsQtyForTheSecondRound;

    /**
     * Конструктор класса.
     * @param emailsQtyForTheFirstRound количество писем, присланных на предварительный тур (раунд).
     * @param emailsQtyForTheSecondRound количество писем, присланных на основной тур (раунд).
     */
    public EmailsCountDigest(long emailsQtyForTheFirstRound, long emailsQtyForTheSecondRound) {
        this.emailsQtyForTheFirstRound = emailsQtyForTheFirstRound;
        this.emailsQtyForTheSecondRound = emailsQtyForTheSecondRound;
    }

    /**
     * Отдаёт количество писем, присланных на предварительный тур (раунд).
     * @return количество писем, присланных на предварительный тур (раунд).
     */
    public long getEmailsQtyForTheFirstRound() {
        return emailsQtyForTheFirstRound;
    }

    /**
     * Отдаёт количество писем, присланных на основной тур (раунд).
     * @return количество писем, присланных на основной тур (раунд).
     */
    public long getEmailsQtyForTheSecondRound() {
        return emailsQtyForTheSecondRound;
    }
}

File: src/main/frontend/src/app/components/answers/answers-list/support/EmailsCountDigest.ts
/*
 * SPDX-License-Identifier: GPL-3.0-only
 * Copyright (c) 2020 - 2022 by Rafael Osipov <rafael.osipov@outlook.com>
 */

export class EmailsCountDigest {
  public static readonly emptyDigest = new EmailsCountDigest(0, 0); // для инициализации в компонентах

  private _emailsQtyForTheFirstRound: number;
  private _emailsQtyForTheSecondRound: number;

  constructor(
    emailsQtyForTheFirstRound: number,
    emailsQtyForTheSecondRound: number
  ) {
    this._emailsQtyForTheFirstRound = emailsQtyForTheFirstRound;
    this._emailsQtyForTheSecondRound = emailsQtyForTheSecondRound;
  }

  get emailsQtyForTheFirstRound(): number {
    return this._emailsQtyForTheFirstRound;
  }

  get emailsQtyForTheSecondRound(): number {
    return this._emailsQtyForTheSecondRound;
  }
}
